model/invoice.rb

```diff
@@ -76,9 +79,26 @@ def charge
       Clog.emit("Invoice couldn't charged with any payment method.") { {invoice_not_charged: {ubid: ubid}} }
       false
     else
+      send_success_email
       true
     end
   end
+
+  def send_success_email(below_threshold: false)
+    ser = Serializers::Web::Invoice.new(:detailed).serialize(self)
+    message = if below_threshold
+      "Since the invoice total of #{ser[:total]} is below our minimum charge threshold, there will be no charges for this month."
+    else
+      "The invoice amount of #{ser[:total]} will be debited from your credit card on file."
+    end
+    Util.send_email(ser[:billing_email], "Ubicloud #{ser[:name]} Invoice ##{ser[:invoice_number]}",
+      greeting: "Dear #{ser[:billing_name]},",
+      body: ["Please find your current invoice ##{ser[:invoice_number]} at the link.",
```

Comment:
How difficult is it to make this attachment?

Reply:
We have several options, but haven't decided what to do yet. I didn't want to spend time on this feature for now. After this PR, only the attachment part remains.
**Possible solutions:**
1. Add a new dependency such as `wkhtmltopdf` and convert our HTML invoices to PDF and send it. We are somewhat conservative about adding new dependencies to our app hosts.
2. (Daniel said it's what Heroku does) Send invoice as an HTML file. We already have an HTML invoice, we can send it. I'm unsure if customers will appreciate HTML invoices
3. Use a pure Ruby library to generate PDF invoices. So we will not add a new dependency, only a new Ruby package

Reply:
3